### PR TITLE
[IMPROVED] Make sure we have decent room in acks since we do not block on insert.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/jetstream_api.go [hunk 1]
```diff
@@ -3755,7 +3755,13 @@ func (s *Server) streamSnapshot(ci *ClientInfo, acc *Account, mset *stream, sr *
 
 	// Create our ack flow handler.
 	// This is very simple for now.
-	acks := make(chan struct{}, 1)
+	ackSize := defaultSnapshotWindowSize / chunkSize
+	if ackSize < 8 {
+		ackSize = 8
+	} else if ackSize > 8*1024 {
+		ackSize = 881024
```
Comment (neilalexander) [inline server/jetstream_api.go:3762]: Is this a typo and should be `8*1024`?

Comment (derekcollison) [reply server/jetstream_api.go:3762]: Ah yes will fix. Thanks.

Comment (derekcollison) [reply server/jetstream_api.go]: Fixed.

**Review by neilalexander — APPROVED** [review]
LGTM

**Review by ripienaar — APPROVED** [review]
LGTM, my tests pass again :)

**Pull request merged**